radix-engine/src/track/track.rs

```diff
@@ -188,12 +196,19 @@ impl<'s, S: SubstateDatabase, M: DatabaseKeyMapper + 'static> Track<'s, S, M> {
                     .and_then(|tracked_node| tracked_node.tracked_partitions.get_mut(&partition))
                 {
                     let db_sort_key = M::to_db_sort_key(&substate_key);
-                    tracked_partition.substates.remove(&db_sort_key);
+                    let tracked_substate = tracked_partition.substates.remove(&db_sort_key);
+                    if let Some(substate) =
+                        tracked_substate.and_then(|s| s.substate_value.into_value())
```

Comment:
Wonder under what scenario the substate is not tracked but marked as transient.

Reply:
There is no scenario unless the system layer is doing something wonky